modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/log/GridLogCommandHandler.java

```diff
@@ -0,0 +1,149 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.rest.handlers.log;
+
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.processors.rest.GridRestCommand;
+
+import org.apache.ignite.internal.processors.rest.GridRestResponse;
+import org.apache.ignite.internal.processors.rest.handlers.GridRestCommandHandlerAdapter;
+import org.apache.ignite.internal.processors.rest.request.GridRestLogRequest;
+import org.apache.ignite.internal.processors.rest.request.GridRestRequest;
+import org.apache.ignite.internal.util.future.GridFinishedFuture;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+import static org.apache.ignite.internal.processors.rest.GridRestCommand.LOG;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Handler for {@link org.apache.ignite.internal.processors.rest.GridRestCommand#LOG} command.
+ */
+public class GridLogCommandHandler extends GridRestCommandHandlerAdapter {
+
+    /**
+     * Supported commands.
+     */
+    private static final Collection<GridRestCommand> SUPPORTED_COMMANDS = U.sealList(LOG);
+    /**
+     * Default log file name *
+     */
+    private static final String DEFAULT_LOG_PATH = "work/log/ignite.log";
+    /**
+     * Default log file start line number *
+     */
+    private static final int DEFAULT_FROM = 0;
+    /**
+     * Default log file end line number*
+     */
+    private static final int DEFAULT_TO = 1;
+    /**
+     * @param ctx Context.
+     */
+    public GridLogCommandHandler(GridKernalContext ctx) {
+        super(ctx);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<GridRestCommand> supportedCommands() {
+        return SUPPORTED_COMMANDS;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IgniteInternalFuture<GridRestResponse> handleAsync(GridRestRequest req) {
+        assert req != null;
+        assert SUPPORTED_COMMANDS.contains(req.command());
```

Comment:
Since only one cmd is supported by this handler there is no need to create SUPPORTED_COMMANDS list.

More optimal is to check that req.command == LOG